app/controllers/categories_controller.rb

```diff
@@ -29,6 +29,9 @@ class CategoriesController < ApplicationController
                      ]
   skip_before_action :verify_authenticity_token, only: %i[search]
 
+  # The front-end is POSTing data to this endpoint, but we're not modifying anything
+  allow_in_readonly_mode :search
```

Comment:
This was hidden in the `read_only_mixin` as a bypass but it feels much better to have it here explicitly (and tested!).